File: SVR.R
# SVR

# Importing the dataset
ds = read.csv('Position_Salaries.csv')
ds = ds[2:3]


# Fitting the SVR Model to the dataset
# install.packages('e1071')
# library(e1071)
lr = svm(formula = Salary ~ ., 
         data = ds,
         type = 'eps-regression')

# Predicting a new result
y_pred = predict(lr, data.frame(Level = 6.5))

# Visualising the SVR Model results
# install.packages('ggplot2')
library(ggplot2)
ggplot() +
  geom_point(aes(x = ds$Level, y = ds$Salary),
             color = 'red') +
  geom_line(aes(x = ds$Level, y = predict(lr, newdata = ds)),
            color = 'purple') +
  ggtitle('SVR Model') +
  xlab('Level') +
  ylab('Salary')

# Visualising the SVR Model results (for higher resolution and smoother curve)
# install.packages('ggplot2')
library(ggplot2)
x_grid = seq(min(ds$Level), max(ds$Level), 0.1)
ggplot() +
  geom_point(aes(x = ds$Level, y = ds$Salary),
             color = 'red') +
  geom_line(aes(x = x_grid, y = predict(lr, newdata = data.frame(Level = x_grid))),
            color = 'purple') +
  ggtitle('SVR Model') +
  xlab('Level') +
  ylab('Salary')

File: SVR.py
# SVR

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
ds = pd.read_csv('Position_Salaries.csv')
X = ds.iloc[:, 1:2].values
Y = ds.iloc[:, 2].values

# Splitting the dataset into the Training set and Test set
"""from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, Y, test_size = 0.2, random_state = 0)"""

# Feature Scaling
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
sc_Y = StandardScaler()
X = sc_X.fit_transform(X)
Y = sc_Y.fit_transform(Y)

# Fitting SVR to the dataset
from sklearn.svm import SVR
lr = SVR(kernel = 'rbf')
lr.fit(X, Y)

# Predicting a new result
y_pred = sc_Y.inverse_transform(lr.predict(sc_X.transform(np.array([[6.5]]))))

# Visualising the SVR results
plt.scatter(X, Y, color = 'red')
plt.plot(X, lr.predict(X), color = 'purple')
plt.title('SVR Model')
plt.xlabel('Position level')
plt.ylabel('Salary')
plt.show()

# Visualising the Regression results (for higher resolution and smoother curve)
X_grid = np.arange(min(X), max(X), 0.1)
X_grid = X_grid.reshape((len(X_grid), 1))
plt.scatter(X, Y, color = 'red')
plt.plot(X_grid, lr.predict(X_grid), color = 'purple')
plt.title('SVR Model')
plt.xlabel('Position level')
plt.ylabel('Salary')
plt.show()